### PR TITLE
Allow incoming data format to be user-definable

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,8 +106,36 @@ The following are available:
...
+type safe. To create a new source format:
+
+1. Create a class, _I_, that will hold all the data for a “row” in the incoming
+data. It must be a class and not an interface.
+1. Create a parameterless method in _I_ for every variable to be exposed. The
+method names must be valid Shemsu names (lowercase with underscores) and
+decorated with `@Export` annotations with the correct type signature. All
+methods must return `boolean`, `long`, `String`, `Instant`, `Set`, or `Tuple`
+(and `Set` and `Tuple` may only contain more of the same).
+1. Create a new interface, _R_, extending `SourceRepository` that fills in the type
```
Comment (callunity) [inline README.md]: I wonder if it makes sense to use `_I_` for the interface? It'll make it slightly easier to keep track of which is the class and which is the interface below.

File: language.md [hunk 2]
```diff
@@ -545,7 +557,8 @@ name spaces. It is possible to create a parameter with the same name as an
 action, though this is not recommended.
 
 ## Variables
-The default variables available in the Shesmu language can be seen on the status page, or:
+The input formats and their variables available in the Shesmu language can be
+seen on the status page, or:
 
     curl http://localhost:8081/variables | js -S .
```
Comment (callunity) [inline language.md]: `jq -S .` ?

Comment (apmasell) [reply language.md]: Yes. Fixed.

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/Export.java [hunk 3]
```diff
@@ -7,8 +7,8 @@
 import java.lang.annotation.Target;
 
 /**
- * Annotation to indicate that a method in {@link Variables} is exported to the
- * Shesmu compiler
+ * Annotation to indicate that a method in {@link GsiStdValue} is exported to
+ * the Shesmu compiler
```
Comment (callunity) [inline shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/Export.java]: Only for `GsiStdValue`, or for all Input Formats?

Comment (apmasell) [reply shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/Export.java]: All of them. Fixed.

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/compiler/OliveNode.java [hunk 4]
```diff
@@ -195,9 +131,12 @@ protected abstract boolean collectDefinitions(Map<String, OliveNodeDefinition> d
 	/**
 	 * Resolve all variable definitions
 	 *
+	 * @param inputFormatDefinition
+	 *            TODO
```
Comment (callunity) [inline shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/compiler/OliveNode.java]: `TODO`?

**Pull request merged**